packages/components/src/combobox-control/index.js

```diff
@@ -94,31 +106,37 @@ export default function ComboboxControl( {
 			</div>
 			<ul { ...menuProps }>
 				{ isOpen &&
-					items.map( ( item, index ) => (
-						// eslint-disable-next-line react/jsx-key
-						<li
-							{ ...getItemProps( {
-								item,
-								index,
-								key: item.key,
-								className: classnames(
-									'components-combobox-control__item',
-									{
-										'is-highlighted':
-											index === highlightedIndex,
-									}
-								),
-								style: item.style,
-							} ) }
-						>
-							{ item === selectedItem && (
-								<Dashicon
-									icon="saved"
-									className="components-combobox-control__item-icon"
-								/>
-							) }
-							{ item.name }
-						</li>
+					( isLoading ? (
+						<span>
+							<Spinner /> { __( 'Loading' ) }
+						</span>
+					) : (
+						items.map( ( item, index ) => (
+							// eslint-disable-next-line react/jsx-key
+							<li
+								{ ...getItemProps( {
+									item,
+									index,
+									key: item.key,
+									className: classnames(
+										'components-combobox-control__item',
+										{
+											'is-highlighted':
+												index === highlightedIndex,
+										}
+									),
+									style: item.style,
+								} ) }
+							>
+								{ item === selectedItem && (
+									<Dashicon
+										icon="saved"
+										className="components-combobox-control__item-icon"
+									/>
+								) }
+								{ item.name }
+							</li>
+						) )
```

Comment:
Can we show an overlay over the current items instead? The current interaction is very jarring.

Reply:
do you mean an overlay for the spinner?

Reply:
An overlay over the list of items instead of the spinner.

Reply:
ah, ok - so the list shows as faded/inactive while the new results are being fetched?

Reply:
Yes

Reply:
👍 will do

Reply:
I added this: the list grays out while the load happens. I noticed the resolved returned an empty list during resolution, so I added some logic in the parent to keep the list unchanged until it resolves (so the previously showing list is greyed out until the fetch completes. Please take a look.

Reply:
removed this bit for now.